web_m2x_options/models/model.py

```diff
@@ -0,0 +1,9 @@
+# -*- coding: utf-8 -*-
+
+from openerp import models, fields
+
+
+class IrModel(models.Model):
+    _inherit = 'ir.model'
+
+    disable_quick_create = fields.Boolean('Disable quick create')
```

Comment:
I don't mind how the field is called, but `base_optional_quick_create` is doing something different so there might be people who need both modules. Of course setting both options at once wouldn't make much sense for the same model, but having both options for different models might be a valid use case. 
This is why I called it `disable_quick_create` to enforce the __disable__ vs. __avoid__ character of this approach.